### ISSUE TITLE
containerd: container not started

### ISSUE THREAD

**Issue by srya-b** [moby/moby#31437]
**Description**
Any docker command would hang and not allow me to interact with the docker daemon at all. So I restarted docker.servce. The restart did not happen cleanly so I had to SIGKILL the docker process and then it was able to stop and restart. Now that docker can communicate with the daemon, I'm unable to start any of the containers that were previously open. The containers sitll show up in `docker ps -a`.

**Steps to reproduce the issue:**
Reproducing the issue is a little tricky as docker had many issues before and the restart wasn't effortless.

**Describe the results you received:**
```
Error response from daemon: containerd: container not started
Error: failed to start containers: <container name>
```
When checking `journalctl -f -u docker.service` the following error message is there:
```
Feb 28 16:53:54 <machine-name> dockerd[18320]: time="2017-02-28T16:53:54.87065004-06:00" level=error msg="containerd: start container" error="containerd: container not started" id=3aa6accd7e1388021396bcac17b5360fcfdc26e99ddd9f5a34dc5a27dfb2273a
Feb 28 16:53:54 <machine-name> dockerd[18320]: time="2017-02-28T16:53:54.871361027-06:00" level=error msg="Create container failed with error: containerd: container not started"
Feb 28 16:53:55 <machine-name> dockerd[18320]: time="2017-02-28T16:53:55.014736480-06:00" level=error msg="Handler for POST /v1.24/containers/<container-name>/start returned error: containerd: container not started"
```

**Describe the results you expected:**
The container would start again.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   78d1802
 Built:        Tue Jan 31 23:35:14 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   78d1802
 Built:        Tue Jan 31 23:35:14 2017
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 44
 Running: 0
 Paused: 0
 Stopped: 44
Images: 278
Server Version: 1.12.6
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 336
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-45-generic
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 32
Total Memory: 125.8 GiB
Name: winterfell
ID: 653G:UPD3:LAAI:ZZME:YWUH:X7SH:WJZV:E3M6:6DVY:SHHJ:JVZG:ZOCA
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

**Comment by sbaks0820** [comment]
I should add that even when I stop the docker service, there are still alot of `containerd-shim` processes that continue to run? I would assume that those processes should stop too when docker stops but idk. Maybe that'll hint at what the problem is.

**Comment by sbaks0820** [comment]
Is it enough to save /var/lib/docker to disk and uninstll docker. wipe all mem of it and re-install it? Will that retain the images and the container contents?

**Comment by thaJeztah** [comment]
ping @mlaventure

**Comment by mlaventure** [comment]
@sbaks0820 which version did you upgrade from?

Do you have the logs when docker was restarted the first time after you upgraded?

Can you share debug log when starting the docker daemon now in that setup?

Thanks.

**Comment by mwaeckerlin** [comment]
I can confirm this strange error message on my system, I can't start some containers any more:

    $ docker ps -a | grep sharing-gallery-php
    91a78e46df77        mwaeckerlin/sharing-gallery   "/bin/sh -c /start.sh"   8 days ago          Exited (1) 21 hours ago                          sharing-gallery-php
    $ docker start sharing-gallery-php
    Error response from daemon: containerd: container not started
    Error: failed to start containers: sharing-gallery-php
    $ lsb_release -d
    Description:    Ubuntu 16.04.2 LTS
    $ docker -v
    Docker version 1.12.6, build 78d1802

There is nothing special in he logs (`docker start` does not write to log).

I do not see any reason, why the container should not start. While looking for this error message, I fount this ticket, which seems to be basically the same problem.

**Comment by mlaventure** [comment]
@mwaeckerlin you probably have rebooted already, but if the issue reproduce itself, could you first temporary enable debug mode in the daemon, then provide the daemon logs?

To enable debug mode without restarting the daemon just do the following:
1. add `"debug": true` to the daemon configuration (by default in `/etc/docker/daemon.json`)
2. reload the configuration with the following: `kill -HUP $(cat /run/docker.pid)`

**Comment by mwaeckerlin** [comment]
Yes, I rebooted and for now the problem is gone. I'll let you know and send the logs, should it happen again.

**Comment by gbitzes** [comment]
Hi, I think I have the same issue. I'm running "docker run -it ubuntu /bin/bash" under Ubuntu 16.10, and get the same error. Here's some logs:

```
sudo[10427]: geobijes : TTY=pts/17 ; PWD=/home/geobijes ; USER=root ; COMMAND=/usr/bin/docker run -it ubuntu /bin/bash
audit[10427]: USER_CMD pid=10427 uid=1000 auid=4294967295 ses=4294967295 msg='cwd="/home/geobijes" cmd=646F636B65722072756E202D6974207562756E7475202F62696E2F62617368 terminal=pts/17 res=success'
audit[10427]: CRED_REFR pid=10427 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/17 res=success'
sudo[10427]: pam_unix(sudo:session): session opened for user root by geobijes(uid=0)
audit[10427]: USER_START pid=10427 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_open acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/17 res=success'
dockerd[2655]: time="2017-03-17T19:59:10.118749608+01:00" level=debug msg="Calling POST /v1.24/containers/create"
dockerd[2655]: time="2017-03-17T19:59:10.118956563+01:00" level=debug msg="form data: {\"AttachStderr\":true,\"AttachStdin\":true,\"AttachStdout\":true,\"Cmd\":[\"/bin/bash\"],\"Domainname\":\"\",\"Entrypoint\":null,\"Env\":[],\"HostConfig\":{\"AutoRemove\":false,\"Binds\":null,\"BlkioDeviceReadBps\":null,\"BlkioDeviceReadIOps\":null,\"BlkioDeviceWriteBps\":null,\"BlkioDeviceWriteIOps\":null,\"BlkioWeight\":0,\"BlkioWeightDevice\":null,\"CapAdd\":null,\"CapDrop\":null,\"Cgroup\":\"\",\"CgroupParent\":\"\",\"ConsoleSize\":[0,0],\"ContainerIDFile\":\"\",\"CpuCount\":0,\"CpuPercent\":0,\"CpuPeriod\":0,\"CpuQuota\":0,\"CpuShares\":0,\"CpusetCpus\":\"\",\"CpusetMems\":\"\",\"Devices\":[],\"DiskQuota\":0,\"Dns\":[],\"DnsOptions\":[],\"DnsSearch\":[],\"ExtraHosts\":null,\"GroupAdd\":null,\"IOMaximumBandwidth\":0,\"IOMaximumIOps\":0,\"IpcMode\":\"\",\"Isolation\":\"\",\"KernelMemory\":0,\"Links\":null,\"LogConfig\":{\"Config\":{},\"Type\":\"\"},\"Memory\":0,\"MemoryReservation\":0,\"MemorySwap\":0,\"MemorySwappiness\":-1,\"NetworkMode\":\"default\",\"OomKillDisable\":false,\"OomScoreAdj\":0,\"PidMode\":\"\",\"PidsLimit\":0,\"PortBindings\":{},\"Privileged\":false,\"PublishAllPorts\":false,\"ReadonlyRootfs\":false,\"RestartPolicy\":{\"MaximumRetryCount\":0,\"Name\":\"no\"},\"SecurityOpt\":null,\"ShmSize\":0,\"UTSMode\":\"\",\"Ulimits\":null,\"UsernsMode\":\"\",\"VolumeDriver\":\"\",\"VolumesFrom\":null},\"Hostname\":\"\",\"Image\":\"ubuntu\",\"Labels\":{},\"NetworkingConfig\":{\"EndpointsConfig\":{}},\"OnBuild\":null,\"OpenStdin\":true,\"StdinOnce\":true,\"Tty\":true,\"User\":\"\",\"Volumes\":{},\"WorkingDir\":\"\"}"
dockerd[2655]: time="2017-03-17T19:59:10.118997608+01:00" level=debug msg="Client and server don't have the same version (client: 1.12.6, server: 1.12.3)"
dockerd[2655]: time="2017-03-17T19:59:10.156407174+01:00" level=debug msg="container mounted via layerStore: /var/lib/docker/aufs/mnt/117fd8f98c1f1a5c34feef838fa643ccff330ad8497ff5186434f0c680424bcf"
dockerd[2655]: time="2017-03-17T19:59:10.186288335+01:00" level=debug msg="Calling POST /v1.24/containers/e0a794b882324c343a3958427d80b819f21467107f6d91137f2452b04dfe8673/attach?stderr=1&stdin=1&stdout=1&stream=1"
dockerd[2655]: time="2017-03-17T19:59:10.186327193+01:00" level=debug msg="Client and server don't have the same version (client: 1.12.6, server: 1.12.3)"
dockerd[2655]: time="2017-03-17T19:59:10.186429716+01:00" level=debug msg="attach: stdin: begin"
dockerd[2655]: time="2017-03-17T19:59:10.186479273+01:00" level=debug msg="attach: stdout: begin"
dockerd[2655]: time="2017-03-17T19:59:10.186525150+01:00" level=debug msg="attach: stderr: begin"
dockerd[2655]: time="2017-03-17T19:59:10.186785601+01:00" level=debug msg="Calling POST /v1.24/containers/e0a794b882324c343a3958427d80b819f21467107f6d91137f2452b04dfe8673/start"
dockerd[2655]: time="2017-03-17T19:59:10.186821473+01:00" level=debug msg="Client and server don't have the same version (client: 1.12.6, server: 1.12.3)"
dockerd[2655]: time="2017-03-17T19:59:10.187561427+01:00" level=debug msg="container mounted via layerStore: /var/lib/docker/aufs/mnt/117fd8f98c1f1a5c34feef838fa643ccff330ad8497ff5186434f0c680424bcf"
dockerd[2655]: time="2017-03-17T19:59:10.187678976+01:00" level=debug msg="Assigning addresses for endpoint backstabbing_wing's interface on network bridge"
dockerd[2655]: time="2017-03-17T19:59:10.187697509+01:00" level=debug msg="RequestAddress(LocalDefault/172.17.0.0/16, <nil>, map[])"
NetworkManager[2405]: <warn>  [1489777150.1892] device (veth8facb07): failed to find device 25 'veth8facb07' with udev
systemd-udevd[10439]: Could not generate persistent MAC address for vethbd46929: No such file or directory
systemd-udevd[10438]: Could not generate persistent MAC address for veth8facb07: No such file or directory
NetworkManager[2405]: <info>  [1489777150.1899] manager: (veth8facb07): new Veth device (/org/freedesktop/NetworkManager/Devices/24)
NetworkManager[2405]: <warn>  [1489777150.1903] device (vethbd46929): failed to find device 26 'vethbd46929' with udev
NetworkManager[2405]: <info>  [1489777150.1909] manager: (vethbd46929): new Veth device (/org/freedesktop/NetworkManager/Devices/25)
audit: ANOM_PROMISCUOUS dev=vethbd46929 prom=256 old_prom=0 auid=4294967295 uid=0 gid=0 ses=4294967295
audit[2825]: SYSCALL arch=c000003e syscall=44 success=yes exit=40 a0=9 a1=c8211f1bf0 a2=28 a3=0 items=0 ppid=1 pid=2825 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="dockerd" exe=2F7573722F62696E2F646F636B657264202864656C6574656429 key=(null)
audit: PROCTITLE proctitle=2F7573722F62696E2F646F636B657264002D480066643A2F2F
kernel: docker0: port 3(vethbd46929) entered blocking state
kernel: docker0: port 3(vethbd46929) entered disabled state
kernel: device vethbd46929 entered promiscuous mode
kernel: IPv6: ADDRCONF(NETDEV_UP): vethbd46929: link is not ready
dockerd[2655]: time="2017-03-17T19:59:10.193430576+01:00" level=debug msg="Assigning addresses for endpoint backstabbing_wing's interface on network bridge"
dockerd[2655]: time="2017-03-17T19:59:10.196699901+01:00" level=info msg="No non-localhost DNS nameservers are left in resolv.conf. Using default external servers : [nameserver 8.8.8.8 nameserver 8.8.4.4]"
NetworkManager[2405]: <info>  [1489777150.1968] devices added (path: /sys/devices/virtual/net/vethbd46929, iface: vethbd46929)
NetworkManager[2405]: <info>  [1489777150.1968] device added (path: /sys/devices/virtual/net/vethbd46929, iface: vethbd46929): no ifupdown configuration found.
NetworkManager[2405]: <info>  [1489777150.1970] devices added (path: /sys/devices/virtual/net/veth8facb07, iface: veth8facb07)
NetworkManager[2405]: <info>  [1489777150.1970] device added (path: /sys/devices/virtual/net/veth8facb07, iface: veth8facb07): no ifupdown configuration found.
dockerd[2655]: time="2017-03-17T19:59:10.197394854+01:00" level=info msg="IPv6 enabled; Adding default IPv6 external servers : [nameserver 2001:4860:4860::8888 nameserver 2001:4860:4860::8844]"
dockerd[2655]: time="2017-03-17T19:59:10.200524304+01:00" level=debug msg="Programming external connectivity on endpoint backstabbing_wing (3938b030aba8928ec4c7ed7ef691b2f6a5564860f6706ef811b29f9c6e2edab5)"
dockerd[2655]: time="2017-03-17T19:59:10.283482035+01:00" level=error msg="containerd: start container" error="containerd: container not started" id=e0a794b882324c343a3958427d80b819f21467107f6d91137f2452b04dfe8673
dockerd[2655]: time="2017-03-17T19:59:10.288656945+01:00" level=error msg="Create container failed with error: containerd: container not started"
dockerd[2655]: time="2017-03-17T19:59:10.288699805+01:00" level=debug msg="attach: stdout: end"
dockerd[2655]: time="2017-03-17T19:59:10.288716389+01:00" level=debug msg="attach: stdin: end"
dockerd[2655]: time="2017-03-17T19:59:10.288863249+01:00" level=debug msg="attach: stderr: end"
dockerd[2655]: time="2017-03-17T19:59:10.288926497+01:00" level=debug msg="Closing buffered stdin pipe"
dockerd[2655]: time="2017-03-17T19:59:10.293420318+01:00" level=debug msg="Revoking external connectivity on endpoint backstabbing_wing (3938b030aba8928ec4c7ed7ef691b2f6a5564860f6706ef811b29f9c6e2edab5)"
audit: ANOM_PROMISCUOUS dev=vethbd46929 prom=0 old_prom=256 auid=4294967295 uid=0 gid=0 ses=4294967295
kernel: docker0: port 3(vethbd46929) entered disabled state
kernel: device vethbd46929 left promiscuous mode
kernel: docker0: port 3(vethbd46929) entered disabled state
NetworkManager[2405]: <info>  [1489777150.3220] devices removed (path: /sys/devices/virtual/net/veth8facb07, iface: veth8facb07)
NetworkManager[2405]: <info>  [1489777150.3220] device (veth8facb07): driver 'veth' does not support carrier detection.
NetworkManager[2405]: <info>  [1489777150.3223] device (vethbd46929): driver 'veth' does not support carrier detection.
NetworkManager[2405]: <info>  [1489777150.3234] devices removed (path: /sys/devices/virtual/net/vethbd46929, iface: vethbd46929)
audit[2825]: SYSCALL arch=c000003e syscall=44 success=yes exit=32 a0=9 a1=c8213bda40 a2=20 a3=0 items=0 ppid=1 pid=2825 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="dockerd" exe=2F7573722F62696E2F646F636B657264202864656C6574656429 key=(null)
audit: PROCTITLE proctitle=2F7573722F62696E2F646F636B657264002D480066643A2F2F
dockerd[2655]: time="2017-03-17T19:59:10.367561199+01:00" level=debug msg="Releasing addresses for endpoint backstabbing_wing's interface on network bridge"
dockerd[2655]: time="2017-03-17T19:59:10.367596875+01:00" level=debug msg="ReleaseAddress(LocalDefault/172.17.0.0/16, 172.17.0.4)"
dockerd[2655]: time="2017-03-17T19:59:10.436091099+01:00" level=error msg="Handler for POST /v1.24/containers/e0a794b882324c343a3958427d80b819f21467107f6d91137f2452b04dfe8673/start returned error: containerd: container not started"
sudo[10427]: pam_unix(sudo:session): session closed for user root
audit[10427]: USER_END pid=10427 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:session_close acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/17 res=success'
audit[10427]: CRED_DISP pid=10427 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/17 res=success'
```

**Comment by mlaventure** [comment]
@gbitzes could you provide the output of `docker-runc --version`?

**Comment by gbitzes** [comment]
Hmm, doesn't look like I have `docker-runc` installed. I get this message when I try to run it:
```
The program 'docker-runc' is currently not installed. You can install it by typing:
sudo apt install docker.io
```

But `docker.io` is already installed on my system... odd.

This is the output of `docker --version`, in case this helps.
```
docker version                                                                                                                                                127 ↵
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   78d1802
 Built:        Tue Jan 31 23:47:34 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Mon, 19 Dec 2016 09:20:48 +1300
 OS/Arch:      linux/amd64
```

**Comment by thaJeztah** [comment]
@gbitzes the `docker.io` package is a package that's built by Ubuntu, not the official packages built here; can you try if installing the official packages works? https://docs.docker.com/engine/installation/linux/ubuntu/

**Comment by jcberthon** [comment]
I had a similar problem but it affected only one container.

My platform is a Raspberry Pi 2 running Ubuntu 16.04 which provides Docker 1.12.6. I wanted to run docker inside Docker (but sharing the docker.sock, not using dind). There is no official image of Docker for armhf, so I downloaded the Dockerfile and docker-entrypoint.sh, and boom I had a Docker image for armhf. However when trying to run it I had the same error message as the others here. And even when activating debug output, I had no better messages.

So I follow @thaJeztah advice and installed latest official Docker, it did not work better **but** the error message was much more explicit: `starting container process caused "exec: \"docker
-entrypoint.sh\": executable file not found in $PATH".`

Oops... I had forgotten to give the excutable bit to the script after I downloaded it :-/ my bad. After changing the permission I can start the container.

So probably other are suffering something similar, their entry point is not executable by the user. Docker 1.12 is unhelpful in providing a good error message to spot that. But the latest version has a very good message!

I hope this helps someone else!
